Fix geo point build error

dstoup opened the pull request:
@aaron-bray 

This patch fixes the ambiguous calls to the geo_point constructor. It was introduced when a second constructor was added. The calls in the serialization test code were creating geo_point objects by just passing {42.50, 73.54} into the first argument. I resolved it by creating actual objects and passing them in. There are 2 issues still. The first is, the call in 

arrows/serialize/json/load_save.cxx

is trying to create a kwiver::vital::geo_point::geo_raw_point_t with just an x,y. The geo_point_raw requires a z value. I added z for the purposes of getting things building, but I'm not sure that's correct.

The other issue is, the 

serialize-protobuf:convert_protobuf.geo_point

test isn't passing. It's possible/likely that it never really worked because it was never really tested but I would like some help making sure it isn't caused by my changes here.

linus-sherrill commented on arrows/serialize/json/load_save.cxx:
```diff
@@ -452,7 +452,7 @@ void load( ::cereal::JSONInputArchive& archive, kwiver::vital::geo_point& point
 
-    const kwiver::vital::geo_point::geo_raw_point_t raw( x, y );
+    const kwiver::vital::geo_point::geo_raw_point_t raw( x, y, 0 );
     point.set_location( raw, crs );
```
The save code needs to save the third coordinate and this code needs to restore it.

linus-sherrill commented on arrows/serialize/json/tests/test_load_save.cxx:
```diff
@@ -127,7 +127,8 @@ TEST( load_save, polygon )
 {
-  kwiver::vital::geo_point obj( {42.50, 73.54}, kwiver::vital::SRID::lat_lon_WGS84 );
+  kwiver::vital::geo_point::geo_2d_point_t geo_2d( 42.50, 73.54 );
+  kwiver::vital::geo_point obj( geo_2d, kwiver::vital::SRID::lat_lon_WGS84 );
```
These tests should test the 3D point

linus-sherrill commented on arrows/serialize/json/tests/test_load_save.cxx:
```diff
@@ -211,7 +212,8 @@ kwiver::vital::metadata create_meta_collection()
-    kwiver::vital::geo_point pt ( { 42.50, 73.54 }, kwiver::vital::SRID::lat_lon_WGS84 );
+    kwiver::vital::geo_point::geo_2d_point_t geo_2d( 42.50, 73.54 );
+    kwiver::vital::geo_point pt ( geo_2d, kwiver::vital::SRID::lat_lon_WGS84 );
     auto* item = info.create_metadata_item( kwiver::vital::any(pt) );
```
This should use the 3D point.

linus-sherrill commented on arrows/serialize/protobuf/tests/test_convert_protobuf.cxx:
```diff
@@ -298,10 +298,12 @@ TEST( convert_protobuf, polygon )
-  kwiver::vital::geo_point obj( {42.50, 73.54}, kwiver::vital::SRID::lat_lon_WGS84 );
+  kwiver::vital::geo_point::geo_2d_point_t geo_2d( 42.50, 73.54 );
+  kwiver::vital::geo_point obj( geo_2d, kwiver::vital::SRID::lat_lon_WGS84 );
 
```
Should test the 3D point

dstoup commented:
Addressed all code review comments. We now store and load geo_raw_point_t in json and protobuf. I've also modified all tests to include a 3d variant.

I haven't modified the serialization code before. It's simple enough (by the looks of it) but it would be good for someone to inspect that code carefully to make sure I haven't missed anything.

dstoup commented:
The CentOS failure is random and not related. I will merge once everything else completes successfully so our builds and PR dashboards will function again.

aaron-bray commented:
This looks good.
Maybe we should rename kwiver::vital::geo_point::geo_raw_point_t to kwiver::vital::geo_point::geo_3d_point_t, to be consistent.

Last I tried, enabling serialization and protobuf on windows does not work :(
So I cannot easily test this... 

Also, in my opinion, there is no need to have a json serializer using cereal
Protobuf reads and writes text in json, as well as their binary format
I don't think there should be 2 serialization implementations, more than likely they will get out of sync and cause issues. And its best to have a schema for reference (i.e. proto files)

dstoup commented:
On Tue, Sep 24, 2019 at 11:04 AM Aaron Bray <notifications@github.com>
wrote:

> This looks good.
> Maybe we should rename kwiver::vital::geo_point::geo_raw_point_t to
> kwiver::vital::geo_point::geo_3d_point_t, to be consistent.
>

I agree. I would like to land this branch first since things are broken but
that should be a quick follow up job. I can probably even take a branch
from my branch, do the work and have it queued for a PR once this lands.


> Last I tried, enabling serialization and protobuf on windows does not work
> :(
> So I cannot easily test this...
>

So, protobuf serialization is disabled on Windows by default, possibly just
on protobuf 2, not sure, but there is some fundemental reason why, I just
can't recall it. The Json stuff does build and run though. I tested that
yesterday on my machine.



> Also, in my opinion, there is no need to have a json serializer using
> cereal
> Protobuf reads and writes text in json, as well as their binary format
> I don't think there should be 2 serialization implementations, more than
> likely they will get out of sync and cause issues. And its best to have a
> schema for reference (i.e. proto files)
>
Interesting though, I will defer to Linus and others about that decision.


—
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Kitware/kwiver/pull/852?email_source=notifications&email_token=ABKEAWTCCIBSXYAA3446RJLQLIUA5A5CNFSM4IZPSFF2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD7OWATA#issuecomment-534601804>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABKEAWXFA5VPTRA4W7ACE2TQLIUA5ANCNFSM4IZPSFFQ>
> .
>


-- 
David Stoup
Principal Engineer

Kitware, Inc.

1712 Route 9, Suite 300
Clifton Park, NY 12065-3104
518-881-4949 (W)
518-312-3946 (M)
518-371-4573 (F)
